### PR TITLE
Update WebAssembly Document (WASM.md).

### PR DESCRIPTION
**Pull request by jplevyak**
Signed-off-by: John Plevyak <jplevyak@gmail.com>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

**Comment by PiotrSikora** [comment]
Why did you close #80? In any case, my comments from there apply here.

File: WASM.md [hunk 1]
```diff
@@ -1,23 +1,64 @@
...
-    - evaluate async emscripten: https://github.com/kripken/emscripten/wiki/Asyncify
-    - evaulate boost::fiber, GNU Pth, getcontext(), setcontext(), swapcontext().
-  - Add support for shared global state.
-  - Intercept logging messages from WAVM and send to Envoy logs.
+. ./emsdk\_env.sh
+
+It is possible later versions will work, e.g.
+./emsdk install latest
+./emsdk activate latest
+However 1.38.25 is known to work.
```
Comment (duderino) [inline WASM.md]: Technically correct, but I can say later versions currently do not work

Comment (jplevyak) [reply WASM.md]: https://github.com/envoyproxy/envoy-wasm/pull/83

I am making up to 1.38.39 work.

**Comment by jplevyak** [comment]
Addressed the comments here.  #80 I though was on the wrong repo because github says:

envoyproxy:master from jplevyak:wasm-docs

which doesn't specify the envoy-wasm repo.  Turns out you have to mouse over... but why ?

**Comment by PiotrSikora** [comment]
Note: This doesn't generate the way you think it does, see: https://github.com/envoyproxy/envoy-wasm/blob/26fa937a9bc713fa17deb57c6bcea512966312db/WASM.md

**Comment by jplevyak** [comment]
Fixed the formatting.  For the record, adding two spaces at the end of the line to force a line break.

**Comment by jplevyak** [comment]
Except that check_format.py doesn't like that.  Sigh.

File: WASM.md (hunk 1 above)
Comment (jplevyak) [reply WASM.md]: #83 has been merged.

File: WASM.md [hunk 2]
```diff
@@ -1,23 +1,68 @@
-Proof of Concept WebAssembly Support.
+WebAssembly Extension Support
 
-# WebAssembly VM
+# Build Dependencies
 
-The WebAssembly VM is https://github.com/WAVM/WAVM. This VM was chosen for the
-proof of concept because:
+## Install glib2.0 if required (on Debian/Ubuntu)
```
Comment (PiotrSikora) [inline WASM.md]: Replace this line with dependency name alone (i.e. `glib2.0`).

File: WASM.md [hunk 3]
```diff
@@ -1,23 +1,68 @@
...
 
-  - It is OSS with a permissive license.
-  - It compiles with LLVM for high performance.
-  - It can attach pre-compiled code to the WASM files (e.g. during distribution
-      from a central controller) for faster startup and reduced load on a pool
-      of proxies.
+apt-get install libglib2.0-dev
 
-# TODO
+## emscripten
```
Comment (PiotrSikora) [inline WASM.md]: `emscripten` is a dependency for regenerating test modules, isn't it?

File: WASM.md [hunk 4]
```diff
@@ -1,23 +1,68 @@
...
+However 1.38.25 is known to work.
+
+## clang-7 or clang-8
+
+export CC=clang  
+export CXX=clang++  
+
+Note: ensure that you have clang in your path (e.g. /usr/lib/llvm-7/bin).
+
+## protobuf v3.6.1
```
Comment (PiotrSikora) [inline WASM.md]: `protobuf` is a dependency for regenerating test modules, isn't it?

File: WASM.md (hunk 3 above)
Comment (jplevyak) [reply WASM.md]: It is also a dependency for building your own WebAssembly module, so it is should be in the main section.

File: WASM.md (hunk 4 above)
Comment (jplevyak) [reply WASM.md]: It is also a dependency for building your own WebAssembly module which uses any protos (e.g. gRPC which is part of the standard API), so it should be included in the main section.

File: WASM.md (hunk 2 above)
Comment (jplevyak) [reply WASM.md]: Done.

**Pull request merged**